Kernel/Syscalls/pledge.cpp

```diff
@@ -14,17 +15,19 @@ ErrorOr<FlatPtr> Process::sys$pledge(Userspace<Syscall::SC_pledge_params const*>
     VERIFY_NO_PROCESS_BIG_LOCK(this);
     auto params = TRY(copy_typed_from_user(user_params));
 
-    if (params.promises.length > 1024 || params.execpromises.length > 1024)
-        return E2BIG;
+    FixedCharBuffer<1024> promises {};
+    bool promises_provided { false };
+    FixedCharBuffer<1024> execpromises {};
```

Comment:
What's the length of the concatenation of all promises (plus a space in between)? That should definitely be below the upper limit.